### ISSUE TITLE
synergyc should fail if connecting to the same Xserver as synergys

### ISSUE THREAD

**Issue by brianjmurrell** [symless/synergy#5508]
If you ```ssh -X client synergyc ...``` where the configuration of the synergy client is to connect to the synergy server on the host you ```ssh```'d from, you will end up having a client and a server on the same X display.

This should be an error and the client should bail out with a useful error message, even perhaps suggesting that an ```ssh -X``` could be responsible as it might not even be obvious that that is the problem if you merely tell the user than the client and server can't connect to the same X display.

**Comment by nlyan** [comment]
Interesting. I take it your objective here is to start Synergy as a client over SSH? Were you using X11 forwarding so you can configure the client using the Synergy GUI?

**Comment by brianjmurrell** [comment]
Yes, the objective is to be able to start the ```synergyc``` client over SSH.  Of course in this case X11 forwarding is not really desired (or maybe it is by people using a GUI -- I don't use it so couldn't care less about X11 forwarding) but it can be easily disregarded/forgotten about and when the client then ends up connecting to the ```synergys``` server's display, it can result in weird artefacts like the mouse warping back onto the ```synergys``` machine's display when it is moved off of the ```synergys``` display on to the ```synergyc``` display.

It becomes an effort to understand why if you have forgotten (or not even realise) that X11 forwarding is enabled and your remote ```synergyc``` is actually driving the display on the ```synergys``` machine.

So, the fix here, as I see it is for the ```synergyc``` to detect that it's actually using the X display that the ```synergys``` is on and bail out with a useful error message, even suggesting that it could be SSH X11 forwarding that is causing the problem.

I can't think of a use-case where anyone would actually want that behaviour anyway.

**Comment by nlyan** [comment]
Brian, so does this work for you if you disable X11 forwarding? 

I'm not sure there's a way to detect if you're trying to use the same X11 display as the server without changing the wire protocol to add send over some kind of X11 specific display ID, and this would have to happen before the client software touched X11 itself.

I'm going to tag this as an enhancement

**Comment by brianjmurrell** [comment]
Yes,  it does all work if I disable X11 forwarding in SSH.  I would expect it would of course.  Assuming you set the ```$DISPLAY``` variable to the display on the remote machine in the remote command and ssh over as the user who has the X11 display open.  So something like:

```
$ ssh client "DISPLAY=:0 synergyc ..."
```
